### ISSUE TITLE
Add another setting for confirming a suggestion

### ISSUE THREAD

**Issue by adamreisnz** [atom/autocomplete-plus#943]
I find that autocomplete is often inserting suggestions when I don't want it to. This happens for example when I've just finished typing a line of code and am about to hit enter, then boom it auto selects a suggestion and I have to go back up and erase it.

Switching the auto complete trigger to TAB doesn't work either, because I use tab often as well and it has the same problem.

Ideally, I would want autocomplete to _never_ insert an autocompletion, unless I explicitly select one by moving arrow keys up or down.

Basically I'd like to see a 5th option in this list:

![image](https://user-images.githubusercontent.com/490562/34758540-cad710aa-f63b-11e7-9e8d-260bc356714c.png)

`Enter, but only when explicitly selected`

I've tried disabling autocomplete altogether at one point because it was driving me crazy, but that didn't help either because I do use it from time to time.

**Comment by adamreisnz** [comment]
Good point, and agree with the suggested change of not having the first row selected by default 👍

**Comment by adamreisnz** [comment]
Anyone willing to draft a PR for this?

It's the single most annoying thing about this package. I use TAB and ENTER for navigating my code _constantly_, and I don't want to use _either_ for confirming an auto complete _unless_ I have explicitly selected a value.

For example, how is the following auto completion useful? I'm wasting so much time on undoing stuff like this:

![ss](https://user-images.githubusercontent.com/490562/41513821-cc3ed590-72f5-11e8-986e-5acc433bdd4e.gif)

cc @benogle @joefitzgerald @nathansobo

**Comment by allefeld** [comment]
Completely agree. Autocomplete keeps getting in the way, no matter how I configure it.

Example: I write CSS, `property:` and hit Tab to get aligned property values. Instead I get some random `::` selector which doesn't even make sense there.

Example: I write text, a list of phrases, and hit Enter to make a new line. Instead the last word gets "completed" by some random word.

If it stays this way I will have to disable it.

I would prefer that a suggestion is only confirmed after it has been explicitly selected. The suggestion by @adamreisnz would work for me. Alternatively, have three options:

- [ ] confirm by Tab
- [ ] confirm by Enter
- [ ] confirm only if explicitly selected
